packages/components/pie-modal/src/index.ts

```diff
@@ -1,18 +1,38 @@
-import { LitElement, html, unsafeCSS } from 'lit'; // eslint-disable-line import/no-extraneous-dependencies
-import { property } from 'lit/decorators.js';
-import { RtlMixin } from '@justeattakeaway/pie-webc-core';
+import { LitElement, unsafeCSS } from 'lit'; // eslint-disable-line import/no-extraneous-dependencies
+import { html, unsafeStatic } from 'lit/static-html.js';
+import { property } from 'lit/decorators.js'; // eslint-disable-line import/no-extraneous-dependencies
+import { RtlMixin, validPropertyValues, requiredProperty } from '@justeattakeaway/pie-webc-core';
 
 import styles from './modal.scss?inline';
+import { HEADING_LEVELS } from './defs';
+
+// Valid values available to consumers
+export { HEADING_LEVELS };
+
+const componentSelector = 'pie-modal';
 
 export class PieModal extends RtlMixin(LitElement) {
     @property({ type: Boolean })
         isOpen = false;
 
-    // eslint-disable-next-line class-methods-use-this
+    @property({ type: String })
+    @requiredProperty(componentSelector)
+        heading: string | undefined;
+
+    @property()
+    @validPropertyValues(componentSelector, Object.values(HEADING_LEVELS), HEADING_LEVELS.H2)
+        headingLevel : HEADING_LEVELS = HEADING_LEVELS.H2;
+
     render () {
+        const {
+            heading,
+        } = this;
+
+        const headingTag = unsafeStatic(this.headingLevel);
+
         return html`
             <dialog class="c-modal" ?open="${this.isOpen}">
-                <h3 class="c-modal-heading">Modal header</h3>
+                <${headingTag} class="c-modal-heading">${heading}</${headingTag}>
```

Comment:
could you potentially do:

```
<h${headingLevel}>${heading}</h${headingLevel>
```

This way you wouldn't need to call `unsafeStatic`?

Reply:
in this example, heading level would be a string number like '1', '2' etc

Reply:
Unfortunately, I believe you still need to use `unsafeStatic` as it's still considered a static expression. I find [the lit doc ](https://lit.dev/docs/templates/expressions/#static-expressions)about the topic quite useful. but in a nutshell, we have a validation layer before using `unsafeStatic ` and only valid heading tags will be passed through with the help of `@validPropertyValues` 

<img width="603" alt="Screenshot 2023-06-09 at 16 50 57" src="https://github.com/justeattakeaway/pie/assets/28605803/21a47649-18a7-454c-8226-084bfe71b6ca">

Reply:
ah, that makes sense. I'm still unsure if I like the idea of a vanilla consumer having to write:

`<pie-modal heading-level="h2" />`

but it's probably fine!

Reply:
It's very common to pass the html tag as a prop as far as I know, tho I would argue the benefits such flexibility will provide to our users, but I assume it was in the requirements for a reason. 

I'm happy to make the prop numerical if it makes more sense and the rest of the team 🚀💯